drivers/i2c/i2c_ll_stm32.c

```diff
@@ -23,12 +23,34 @@ static int i2c_stm32_runtime_configure(struct device *dev, u32_t config)
 	const struct i2c_stm32_config *cfg = DEV_CFG(dev);
 	struct i2c_stm32_data *data = DEV_DATA(dev);
 	I2C_TypeDef *i2c = cfg->i2c;
-	u32_t clock;
+	u32_t clock = 0;
 
-	data->dev_config = config;
+#if defined(CONFIG_SOC_SERIES_STM32F3X) || defined(CONFIG_SOC_SERIES_STM32F0X)
+	switch ((u32_t)cfg->i2c) {
```

Comment:
I feel this code would be nice if clock source source was a priori unknown, though, since it is not the case here, and hence it sounds a bit "over featured".
Isn't there an alternate solution to get SYSCLK value (whatever the instance)?

Reply:
We could use the LL_RCC_GetSystemClocksFreq(). I tried it but the code crashes inside this function.

Reply:
done